### PR TITLE
Resolved max and min bifurcation via the word optimal

### PR DESCRIPTION
**Pull request by tbrysiewicz**
Simplification:
maximal_vertex(LP)/minimal_vertex(LP) -> optimal_vertex(LP)
maximal_value(LP)/minimal_value(LP) -> optimal_value(LP)
where optimal_* retrieves the max/min convention of the linear program

### CODE REVIEW

File: src/PolyhedralGeometry/LinearProgram.jl [hunk 1]
```diff
@@ -96,45 +96,11 @@ feasible_region(lp::LinearProgram) = lp.feasible_region
...
 """
-function minimal_vertex(lp::LinearProgram)
-   mv = lp.polymake_lp.MINIMAL_VERTEX
-   if mv != nothing
-      return dehomogenize(mv)
-   else
-      return nothing
-   end
-end
+    optimal_vertex(LP::LinearProgram)
```
Comment (fingolfin) [inline src/PolyhedralGeometry/LinearProgram.jl:101]: Not to nitpick, but have you considered the term `extremal_nitpick`? AFAIK "extremal = minimal or maximal" is a very common convention in math. not so sure about "optimal"...

Comment (micjoswig) [reply src/PolyhedralGeometry/LinearProgram.jl:101]: That is how these functions are called in standard optimization software.  So, this choice is superior to "extremal".

**Comment by alexej-jordan** [comment]
Code looks good and this semantic change makes sense. You still need to adjust the docs though.

Do we care about a least optimal value/vertex? If so, accessing it is now harder, i.e. for `LP::LinearProgram`you have to do
```
c, k = objective_function(LP)
LP_inv = LinearProgram(feasible_region(LP), c; k = k, convention = :min)
```
and then call `optimal_*(LP_inv)`.

**Comment by tbrysiewicz** [comment]
> Code looks good and this semantic change makes sense. You still need to adjust the docs though.
> 
> Do we care about a least optimal value/vertex? If so, accessing it is now harder, i.e. for `LP::LinearProgram`you have to do
> 
> ```
> c, k = objective_function(LP)
> LP_inv = LinearProgram(feasible_region(LP), c; k = k, convention = :min)
> ```
> 
> and then call `optimal_*(LP_inv)`.

Perhaps a function that does that? What's the best name? @micjoswig  
```
LP_inv=opposite(LP::LinearProgram)
```

**Comment by micjoswig** [comment]
> > Do we care about a least optimal value/vertex?

No, we don't.  Geometrically this would be kind of meaningful, but from the optimization perspective that's totally unnatural.  So, don't bother.

**Pull request merged**